### ISSUE TITLE
crash with panic contiuously on MacBook Pro 2008

### ISSUE THREAD

**Issue by duzun** [syncthing/syncthing#5076]
I'm using syncthing on Linux and OSx (with QSyncthing v0.5.8).
I keep my personal folders in sync (Desktop, Pictures, Video, Downloads, Music, Documents etc) and a bunch of other folders with lots of files, syncing to multiple machines, including a copy on a Debian server.

Everything works fine on every system except OSx. I managed to figure out that the DB gets corrupted and I did try to remove it and start syncthing again. It works for a while (5-10 minutes), starts scanning all folders, fan runs like crazy, then crash. I did try it again in few days, several times. I even reinstalled syncthing in the hope it would finish scanning all folders and not crash, but it still happens again and again. Right now I have a collection of 97 panic logs for last half an hour.

Here is the piece where it crashes:

```log
panic: leveldb/table: corruption on data-block (pos=1871238): checksum mismatch, want=0x701d7173 got=0xe7442995 [file=003484.ldb]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x451955c]
```

It did work fine for a while though, I guess until I reached a certain threshold on number/size of files and folders.

### Version Information

Syncthing Version: v0.14.49 "Dysprosium Dragonfly" (go1.10.3 darwin-amd64) teamcity@build.syncthing.net 2018-07-10 15:40:06 UTC
OS Version: OS X El Capitan v10.11.6
Machine: MacBook Pro 15-inch, Late 2008

**Comment by calmh** [comment]
Your database is toast, due to disk corruption, memory corruption, or potentially a software bug (although as far as we've been able to determine, not one in Syncthing). You will unfortunately need to remove the database and reindex. The forum can assist if you need help!

**Comment by duzun** [comment]
Thanks for answer.

I'll run memcheck, disk check and clean DB once again to see if that helps.
I see no other serious issues with my system.
I also have Dropbox, Google Backup & Sync and they run with no issues.

**Comment by duzun** [comment]
I did run disk check, removed DB, stopped all running programs on Mac, then started syncthing and left it for the night. The next day it have managed to finish scaning!

I did one more thing:
I've installed Linux on Mac with dual-boot, shared home folder and UID of user on both systems, made ~/.config/syncthing a symlink to ~/Library/Application\ Support/Syncthing and installed syncthing on linux. Works just great! No more lagging and panics.

Not related to the issue, but I thought its a cools story to share :-)
